docs/sources/flow/reference/components/discovery.ovhcloud.md

```diff
@@ -0,0 +1,165 @@
+---
+aliases:
+- /docs/grafana-cloud/agent/flow/reference/components/discovery.ovhcloud/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/discovery.ovhcloud/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/components/discovery.ovhcloud/
+- /docs/grafana-cloud/send-data/agent/flow/reference/components/discovery.ovhcloud/
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/discovery.ovhcloud/
+description: Learn about discovery.ovhcloud
+title: discovery.ovhcloud
+---
+
+# discovery.ovhcloud
+
+`discovery.ovhcloud` discovers scrape targets from OVHcloud's [dedicated servers][] and [VPS][] using their [API][]. 
+{{< param "PRODUCT_ROOT_NAME" >}} will periodically check the REST endpoint and create a target for every discovered server. 
+The public IPv4 address will be used by default - if there's none, the IPv6 address will be used. 
+This may be changed via relabeling with `discovery.relabel`. 
+For OVHcloud's [public cloud][] instances you can use `discovery.openstack`.
+
+[API]: https://api.ovh.com/
+[public cloud]: https://www.ovhcloud.com/en/public-cloud/
+[VPS]: https://www.ovhcloud.com/en/vps/
+[Dedicated servers]: https://www.ovhcloud.com/en/bare-metal/
+
+## Usage
+
+```river
+discovery.ovhcloud "LABEL" {
+    application_key    = APPLICATION_KEY
+    application_secret = APPLICATION_SECRET
+    consumer_key       = CONSUMER_KEY
+    service            = SERVICE
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name               | Type           | Description                                                    | Default       | Required
+------------------ | -------------- | -------------------------------------------------------------- | ------------- | --------
+endpoint           | `string`       | [API][] endpoint.                                              | "ovh-eu"      | no
```

Comment:
Personal nit; feel free to disregard.

I find these easier to read if they're listed in order, first the required field, then the optional ones.

Reply:
The latest consensus is to sort these alphabetically: https://github.com/grafana/agent/pull/5845#discussion_r1404487210

TBH I didn't try to do it here because I wasn't sure how to do it easily. I just tried it with Copilot. It worked, and by chance it turns out the optional fields got grouped together, almost at the end of the table :)

Reply:
😅 didn't know about the convention, thanks for letting me know!